sample_size/multiple_testing.py

```diff
@@ -0,0 +1,67 @@
+from typing import List
+
+import numpy as np
+from statsmodels.stats.multitest import multipletests
+
+from sample_size.metrics import BaseMetric
+
+
+class MultipleTestingMixin:
+    """
+    This class calculates sample size required under the case of multiple testing
+
+    Attributes:
+    metrics: a list of BaseMetric registered by users
+    variants: number of variants, including control
+    alpha: statistical significance
+    power: statistical power
+
+    """
+
+    metrics: List[BaseMetric]
+    alpha: float
+    power: float
+    variants: int
+
+    REPLICATION: int = 500
+
+    def get_multiple_sample_size(self, lower: float, upper: float, depth: int = 0) -> int:
+        max_recursion_depth: int = 20
+        epsilon: float = 0.025  # TODO(any): make this configurable by users ML-5429
+
+        if depth > max_recursion_depth:
+            raise RecursionError(f"Couldn't find a sample size that satisfies the power you requested: {self.power}")
+
+        candidate = int(lower + (upper - lower) / 2)
+        expected_power = self._expected_average_power(candidate)
+        if np.isclose(self.power, expected_power, epsilon) or (lower == upper and self.power < expected_power):
+            return candidate
```

Comment:
I added the condition `lower == upper and self.power < expected_power` because in some cases like https://github.com/godaddy/sample-size/pull/32#pullrequestreview-995494156, the lower bound sample size generates average power way above 0.8. In this case we do want to use lower bound, i.e. the maximum of individual sample sizes, instead of raising a runtime error

Reply:
Oh! I see the bug! I think we proved in the simulation studies that BH uniformly decreases power (by definition it can't lower it because it is conservative, and never decreases p-values), so we shouldn't never run into the problem you're describing, but I think

```python
        lower = max([self._get_single_sample_size(metric, self.alpha) for metric in self.metrics])
```

should be

```python3
        lower = min([self._get_single_sample_size(metric, self.alpha) for metric in self.metrics])
```

I suspect that would fix the lower bound still being > 0.8 problem

Reply:
It did fix the problem!